File: number guessing game/NumberGuessingGame.py
import random

lower, upper = input("Enter range: ").split()
lower, upper = int(lower), int(upper)
n = 10
systemNumber = random.randint(lower, upper + 1)
userNumber = int(input(f"Enter a number in the range {lower} to {upper}: "))
while n != 0:
    if userNumber > upper or userNumber < lower:
        userNumber = int(input(f"You're going out of bound! Try again within the range {lower} to {upper}: "))
    else:
        if userNumber == systemNumber:
            print("Congratulations! You guessed it right!")
            break
        elif userNumber > systemNumber:
            userNumber = int(input("Too high! Try again! Enter your guess: "))
        else:
            userNumber = int(input("Too low! Try again! Enter your guess: "))
    if n == 1:
        print("Sorry, you've exhausted all your trials! :(")
    n -= 1

File: number guessing game/NumberGuessingGame.java
package projectzz;

import java.util.*;

public class NumberGuessingGame {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter range: ");
        int lower = sc.nextInt();
        int upper = sc.nextInt();
        int n = 10;
        int systemNumber = (int) (Math.random() * (upper - lower)) + lower;
        System.out.print("Enter a number in the range " + lower + " to " + upper + ": ");
        int userNumber = sc.nextInt();
        while (n != 0) {
            if (userNumber > upper || userNumber < lower) {
                System.out.print("You're going out of bound! Try again within the range " + lower + " to " + upper + ": ");
                userNumber = sc.nextInt();
            }
            else {
                if (userNumber == systemNumber) {
                    System.out.println("Congratulations! You guessed it right!");
                    break;
                }
                else if (userNumber > systemNumber) {
                    System.out.print("Too high! Try again! Enter your guess: ");
                    userNumber = sc.nextInt();
                }
                else {
                    System.out.print("Too low! Try again! Enter your guess: ");
                    userNumber = sc.nextInt();
                }
            }
            if (n == 1) {
                System.out.println("Sorry, you've exhausted all your trials! :(");
            }
            n--;
        }
    }
}
